### ISSUE TITLE
The defaults object does not respect the forceUpdate feature

### ISSUE THREAD

**Issue by jpodwys** [jpodwys/superagent-cache#48]
[This function](https://github.com/jpodwys/superagent-cache/blob/master/utils.js#L176) ignores the `forceUpdate` value provided to it.

**Comment by jpodwys** [comment]
@vijayst please try out the forceupdate-defaults branch and let me know if it works better for you than before.

**Comment by vijayst** [comment]
I used the branch forceupdate-defaults. The refresh part is working but the cache is not working. To help clarify, I will post some code here.

```
import superagentCache from "superagent-cache";

const superagent = superagentCache();

API.config({
  urlRoot: API_ROOT,
  superagent,
  requestHeaders: {
    "X-DreamFactory-Api-Key": APP_TOKEN,
    "Content-Type": "application/json; charset=utf-8",
  },
});
```
The API config passes defaults for some of the superagent params. In this, I pass the superagent patched by superagent-cache.

In my refresh method of models, I have some code to do refresh (this is the only place I want to invalidate the cache or force AJAX call).
```
refresh () {
    API.requestHeaders[HEADER_SESSION_TOKEN] = authProvider.fetchToken();
    const superagent = superagentCache(null, null, { forceUpdate: true });
    return API.request({
      superagent,
      endpoint: `${ENDPOINT}/${this.id}?related=` +
      `contract_type_by_contract_type_id`,
       onSuccess: (response) => {
        this.set({ modelJson: response.body });
        superagentCache();
      },
      onError: (err) => {
        superagentCache();
        throw err;
      }
    });
  }
```
In the above method, I pass a superagent (with forceUpdate set to true). After the call completes, in onSuccess and onError callbacks, I am hoping to re-patch superagent with default params.

The work-around that I was planning to use is:
- Do not pass default superagent in the API.config call.
- Pass superagentCache() in all GET calls except refresh.

Since mobx-model has its own superagent, the default superagent will take care of all the calls except the GET calls for which I will be sending a superagent which I have patched with superagent-cache.

**Comment by jpodwys** [comment]
OK, here's what I believe you're trying to do:

1. Pass in a patched superagent instance to mobx-model with `forceUpdate` set to `true`
2. Use a refresh function that, upon completion, resets the defaults so `forceUpdate` is `false`
3. Continue this cycle such that all mobx-model calls use `forceUpdate` but then set `forceUpdate` to `false` upon finishing ajax calls

Here are some notes:

You're trying to reset superagent-cache's `defaults` by simply calling `superagentCache()`. That will not work. The only way to update superagent-cache's defaults once it has been initialized is via the `defaults` property as follows: `superagent.defaults = {}`;

This likely explains why "the refresh part is working but the cache is not working." It appears that the cache is not working because you're resetting the defaults incorrectly. Therefore, all queries are using `forceUpdate`.

This means that you would need to set `superagent.defaults = {forceUpdate: true};` just before every mobx-model ajax call and `superagent.defaults = {};` just before every other ajax call.

However, I do not recommend this. Because `superagent.defaults` is global to all superagent queries, it's possible you'll end up with an unexpected `forceUpdate` value if a mobx-model query and a custom query fire off at the same time.

Instead, I recommend this: since I fixed the bug where `forceUpdate` was not consumed as part of the `defaults` object, I don't understand why you can't just pass the result of the following line of code into mobx-model:

```javascript
var superagent = superagentCache(null, null, { forceUpdate: true });
```

so that mobx-model can enjoy the benefit of `forceUpdate` while doing your custom queries as follows:

```javascript
superagent
  .get('uri')
  .forceUpdate(false)
  .end(function(err, response){
    // Handle response
  }
)
```

**Comment by vijayst** [comment]
I could do it. Leave alone mobx-model the way it is. Don't pass any superagent to it. Use superagent cache for my project and use it in queries I want to cache. 

I was hoping that mobx-model would wrap superagent-cache. I wanted a cache param in mobx-model config which will use superagent-cache. I want a forceUpdate param in each request. 

Unfortunately, we ended up with a solution where we pass superagent object in config and another superagent object in each request (which will override the config).

**Comment by jpodwys** [comment]
That's not the same thing I just recommended. I recommended passing a patched superagent instance into mobx-model and then adding the `.forceUpdate(false)` chainable to your custom queries.

**Comment by vijayst** [comment]
mobx-model wraps the superagent calls. 
`API.request({ params })` makes the call to `superagent.get().end()`. As of now, mobx-model does not recognize forceUpdate param. The API returns the promise of the end() call. So, I don't have access to superagent to chain the forceUpdate method call.

**Comment by jpodwys** [comment]
The bug reported in this issue is that the `forceUpdate` setting was not evaluated as part of the `defaults` object. That issue is fixed in `1.7.1`.

As for your issue with mobx-model wrapping superagent, I'm unable to help. You'll need to speak with the maintainers of mobx-model if you'd like them to change anything.

Thanks again for the bug report concerning `forceUpdate` and good luck with mobx-model.
